app/controllers/oai_controller.rb

```diff
@@ -93,23 +108,62 @@ def identify_params
     oai_params(:verb)
   end
 
-  # rubocop:disable Metrics/AbcSize
-  def list_record_normalized_dump_candidates
-    @list_record_normalized_dump_candidates ||= begin
-      current_full_dump = @stream.current_full_dump
+  # Get a page of OAI-XML records and a token pointing to the next page
+  def next_record_page(token = nil)
+    # parse the token if we were provided one
+    set, page, from_date, until_date = OaiConcern::ResumptionToken.decode(token) if token
```

Comment:
It's a little funny we're encoding a resumption token (in the caller) in order to immediately decode it (here). Maybe `OaiConcern::ResumptionToken` would be easier to work with if we made it an actual class with attributes instead of a container for class-level methods? Maybe something like:

```
class OaiResumptionToken
  attr_reader :set, :page, :from_date, :until_date

  def initialize; end
  def encode; end
  def valid?; end
end
```

Reply:
Implemented this and pushed a little bit more error handling on this class.